api/config/config_test.json

```diff
@@ -0,0 +1,5 @@
+{
```

Comment:
Can we rename this from `config_test` to something more descriptive? (Or if it is just for test purposes, can we comment what the purpose is?)

Reply:
This is a configuration file with settings for the "test" environment. This will probably get clearer when we have similarly files for "staging", "prod", etc.

Reply:
Ah, okay, I think that will clear things up some. I read it as a file created to test the config functionality, rather than a config for the test environment. Would it be possible to have it follow the same naming convention as our environment files? I'm not actually certain if it would work, but it could make things more clear? (`environment.prod.ts`, `environment.test.ts` and `environment.ts`?)

Reply:
I agree the filenames will become self-explanatory as we get more of them.

Not being able to have comments in JSON files has been frustrating in the RDR, though. Could we store configs in YAML or (ugh) XML or ASCII protos or something that supports comments? (We could add bogus "comment":"my comment text" values, but that would either clutter our db format or break Dan's validation logic.)

Reply:
I like the idea of being able to add comments; I think if we had that, the filename wouldn't matter as much.

JSON is convenient in terms of support for parsing into objects based on reflection (i.e. Gson), formatting, is easy to read, etc. Comments are a weak point. I could potentially add a preprocessing step in ConfigLoader which strips out lines that start with  // or # or whatever... though then IDEs that work with JSON may complain. WDYT?

I'm not sure about parsing based on reflection for YAML or XML. There's probably some stuff out there.

Protos in text format are an interesting option as they'd handle all of the above. I could look into switching to that if you think it's worthwhile.

Let me know what y'all think.

Reply:
I chatted with Mark. I think I'm going to stick with JSON, and just strip out comments (lines starting with //) before loading. Stay tuned.